File: app/src/main/java/com/example/data/PricingCalculator.kt
package com.example.data

/**
 * Subscription-tier consultation pricing, extracted from the ViewModel so the
 * discount rules can be unit tested without an Android/Room context.
 */
object PricingCalculator {
    private val bronzeFreeAstrologerIds = setOf(1, 3, 6)
    private val silverFreeAstrologerIds = setOf(1, 3, 4, 6, 7)

    fun getConsultationPrice(astrologerId: Int, basePrice: Int, tier: String): Int {
        return when (tier) {
            "Mahadasha Gold" -> 0
            "Rashifal Silver" -> {
                if (astrologerId in silverFreeAstrologerIds) 0 else (basePrice * 0.75).toInt()
            }
            "Nakshatra Bronze" -> {
                if (astrologerId in bronzeFreeAstrologerIds) 0 else (basePrice * 0.90).toInt()
            }
            else -> basePrice
        }
    }
}

File: web/src/lib/pricing-calculator.ts
/**
 * Subscription-tier consultation pricing. Ported from the Android app's
 * PricingCalculator.kt so the discount rules stay identical across platforms.
 */

const BRONZE_FREE_ASTROLOGER_IDS = new Set([1, 3, 6]);
const SILVER_FREE_ASTROLOGER_IDS = new Set([1, 3, 4, 6, 7]);

export function getConsultationPrice(
  astrologerId: number,
  basePrice: number,
  tier: string
): number {
  switch (tier) {
    case "Mahadasha Gold":
      return 0;
    case "Rashifal Silver":
      return SILVER_FREE_ASTROLOGER_IDS.has(astrologerId) ? 0 : Math.floor(basePrice * 0.75);
    case "Nakshatra Bronze":
      return BRONZE_FREE_ASTROLOGER_IDS.has(astrologerId) ? 0 : Math.floor(basePrice * 0.9);
    default:
      return basePrice;
  }
}
